### PR TITLE
 git HOOK: add first 'pre-push' hook to check CHANGELOG

### PR DESCRIPTION
**Pull request by egypcio**
- verify if the **CHANGELOG.md** exists;
  - make sure we tagged a release before pushing;
  - the changelog file must contain a date;
  - date format defined by https://pkidd.com/date.htm

### CODE REVIEW

**Comment by bkircher** [comment]
rebased

**Comment by bkircher** [comment]
Hey hey :wave:  

Shouldn't this be in `.git/hooks` according to GITHOOKS(5) ?

File: .githooks/pre-push/check-changelog.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+############################################################
+CHANGELOG="CHANGELOG.md"
+has_CHANGELOG=$($GIT_CMD ls-tree --full-tree -r --name-only HEAD | grep -i $CHANGELOG)
+if [ ! -z $has_CHANGELOG ]; then
+  ( grep $GIT_TAG $CHANGELOG && \
+    echo $GIT_TAG_COMMIT | grep -v $GIT_COMMIT_SHORT ) >/dev/null && exit 0 || exit 123
+  ( grep "${DATE}" $CHANGELOG ) || exit 321
+else
+  echo "Please tag this release, and rework CHANGELOG.md to add a date."
+  exit 666
```
Comment (bkircher) [inline .githooks/pre-push/check-changelog.sh:21]: Let simplify this a bit.

Can you change the error message to "No $CHANGELOG found". Can you make the error exit codes 1, 2, 3 or something. No magic

**Review by bkircher — DISMISSED** [review]

File: .githooks/pre-push/check-changelog.sh (hunk 1 above)
Comment (bkircher) [reply .githooks/pre-push/check-changelog.sh:21]: In general, exit codes should be between 0-255. I The devils numbers won't fit into an unsigned byte :smile: :imp:

Comment (bkircher) [reply .githooks/pre-push/check-changelog.sh:21]: Use -n instead of ! -z for the test. Makes it a bit simpler, too.

**Comment by egypcio** [comment]
> Hey hey 👋
> 
> Shouldn't this be in `.git/hooks` according to GITHOOKS(5) ?

https://git-hooks.github.io/git-hooks and https://github.com/git-hooks/git-hooks do use a do document the use of `.githook`

**Comment by bkircher** [comment]
Thanks, looks good. One last thing, though. Okay for you if we drop line 3? I think we should assume PATH is correctly set on the system when this script is executed and refrain from altering it.

**Comment by egypcio** [comment]
> Thanks, looks good. One last thing, though. Okay for you if we drop line 3? I think we should assume PATH is correctly set on the system when this script is executed and refrain from altering it.

I am fine with that. it's just an old habit ;3
committed and pushed already. ready to merge

**Review by bkircher — APPROVED** [review]

**Comment by bkircher** [comment]
Thanks!

**Pull request merged**